repo2docker/buildpacks/conda/activate-conda.sh

```diff
@@ -3,9 +3,16 @@ CONDA_PROFILE="${CONDA_DIR}/etc/profile.d/conda.sh"
 test -f $CONDA_PROFILE && . $CONDA_PROFILE
 if [[ "${KERNEL_PYTHON_PREFIX}" != "${NB_PYTHON_PREFIX}" ]]; then
     # if the kernel is a separate env, stack them
-    # so both are on PATH
+    # so both are on PATH, notebook first
     conda activate ${KERNEL_PYTHON_PREFIX}
     conda activate --stack ${NB_PYTHON_PREFIX}
+
+    # even though it's second on $PATH
+    # make sure CONDA_DEFAULT_ENV is the *kernel* env
+    # so that `!conda install PKG` installs in the kernel env
+    # where user packages are installed, not the notebook env
+    # which only contains UI when the two are different
```

Comment:
I had to read this a few times for it to make sense but I can't suggest a different wording that is easier to parse :-/

You always want to install into the "kernel env" and never into the "notebook env", but sometimes the two are the same in which case you are still installing into the kernel env, it just happens to also be the notebook env. You just gotta have your wits about you with all these envs :)